Source/DafnyCore/DafnyPrelude.bpl

```diff
@@ -649,7 +649,7 @@ procedure $IterHavoc1(this: ref, modi: Set Box, nw: Set Box);
 procedure $IterCollectNewObjects(prevHeap: Heap, newHeap: Heap, this: ref, NW: Field (Set Box))
                         returns (s: Set Box);
   ensures (forall bx: Box :: { s[bx] } s[bx] <==>
-              read(newHeap, this, NW)[bx] ||
+              (read(newHeap, this, NW) : Set Box)[bx] ||
```

Comment:
This is not needed, but because you will eventually remove the alpha from read, you add it preventively, right?

Reply:
Exactly. And the code in Dafny that generates Boogie code does this for some but not all instances of `read`, as well. Once it does it everywhere, we can remove the type parameter to `Field`.